### PR TITLE
Part8 all tasks done

### PR DESCRIPTION
**Pull request by alexkulesh**

### CODE REVIEW

File: FirstSolution/Part8Task3P1/Task.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+            int n = 1000;
+            int i = 0;
+
+            while (n >= 50)
+            {
+                n /= 2;
+                i++;
+            }
+            
+            Console.WriteLine("Количество итераций равна - " + i);
```
Comment (Alex19777) [inline FirstSolution/Part8Task3P1/Task.cs:25]: Write result after dividing

File: FirstSolution/Part8Task3P2/Task.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+            int number = 1000;
+            int i = 0;
+
+            for (int j = 0; number >= 50; j++)
+            {
+                number /= 2;
+                i++;
+            }
+
+            Console.WriteLine("Количество итераций равна - " + i);
```
Comment (Alex19777) [inline FirstSolution/Part8Task3P2/Task.cs:25]: Same

File: FirstSolution/Part8Task3P2/Task.cs [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    // Дано число n=1000. Делите его на 2 столько раз, пока результат
+    // деления не станет меньше 50. Какое число получится? Посчитайте
+    // количество итераций, необходимых для этого (итерация - это проход цикла).
+    // Используйте цикл for.
+
+    public class Task
+    {
+        static void Main(string[] args)
+        {
+            int number = 1000;
```
Comment (Alex19777) [inline FirstSolution/Part8Task3P2/Task.cs]: 'n' or 'number'?

File: FirstSolution/Part8Task4P1/Task.cs [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    // Пользователь вводит число. Выведите n строк, добавляя следующее
+    // натуральное число на каждой строке. Используйте цикл while
+    
+    public class Task
+    {
+        static void Main(string[] args)
+        {
+            var n = Convert.ToInt32(Console.ReadLine());
+            int i = 0;
+            string num = n.ToString();
```
Comment (Alex19777) [inline FirstSolution/Part8Task4P1/Task.cs]: Simplify. Use n instead of num

File: FirstSolution/Part8Task4P1/Task.cs [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+            string t = "";
+            
+            while (i <= num.Length - 1)
+            {
+                t += num[i];
```
Comment (Alex19777) [inline FirstSolution/Part8Task4P1/Task.cs]: Don't you get any out of range exception here?

File: FirstSolution/Part8Task5GuessTheNumber/Task.cs [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    // сыграть несколько раз без перезапуска программы. По желанию разработчика можно предусмотреть
+    // вариант с ограничением количества попыток пользователя для угадывания числа.
+
+    public class Task
+    {
+        static void Main(string[] args)
+        {
+            Random rnd = new Random();
+            int number = rnd.Next(0, 100);
+            Console.WriteLine(number);
```
Comment (Alex19777) [inline FirstSolution/Part8Task5GuessTheNumber/Task.cs]: Remove this random number before you will try to guess the number

File: FirstSolution/Part8Task5GuessTheNumber/Task.cs [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+            for (int i = 1; i <= 3; i++)
```
Comment (Alex19777) [inline FirstSolution/Part8Task5GuessTheNumber/Task.cs]: Get the attempts number from user input

File: FirstSolution/Part8Task5GuessTheNumber/Task.cs [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+                    Console.WriteLine("Меньше");
+                }
+                else if (input > number)
+                {
+                    Console.WriteLine("Больше");
+                }
+                else
+                {
+                    Console.WriteLine("Некорректно введены данные");
+                }
```
Comment (Alex19777) [inline FirstSolution/Part8Task5GuessTheNumber/Task.cs]: Do you really need this else here?

File: FirstSolution/Part8Task3P1/Task.cs (hunk 1 above)
Comment (alexkulesh) [reply FirstSolution/Part8Task3P1/Task.cs:25]: Done

File: FirstSolution/Part8Task3P2/Task.cs (hunk 2 above)
Comment (alexkulesh) [reply FirstSolution/Part8Task3P2/Task.cs:25]: Done

File: FirstSolution/Part8Task3P2/Task.cs (hunk 3 above)
Comment (alexkulesh) [reply FirstSolution/Part8Task3P2/Task.cs]: Considered and changed

File: FirstSolution/Part8Task4P1/Task.cs (hunk 4 above)
Comment (alexkulesh) [reply FirstSolution/Part8Task4P1/Task.cs]: Done

File: FirstSolution/Part8Task4P1/Task.cs (hunk 5 above)
Comment (alexkulesh) [reply FirstSolution/Part8Task4P1/Task.cs]: Changed

File: FirstSolution/Part8Task5GuessTheNumber/Task.cs (hunk 6 above)
Comment (alexkulesh) [reply FirstSolution/Part8Task5GuessTheNumber/Task.cs]: Removed

File: FirstSolution/Part8Task5GuessTheNumber/Task.cs (hunk 7 above)
Comment (alexkulesh) [reply FirstSolution/Part8Task5GuessTheNumber/Task.cs]: Done

File: FirstSolution/Part8Task5GuessTheNumber/Task.cs (hunk 8 above)
Comment (alexkulesh) [reply FirstSolution/Part8Task5GuessTheNumber/Task.cs]: Changed

**Review by Alex19777 — APPROVED** [review]

**Review by Alex19777 — APPROVED** [review]

**Review by Alex19777 — APPROVED** [review]

**Pull request merged**